tests/vbank/v_bank.go

```diff
@@ -0,0 +1,839 @@
+package vbank
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ngaut/log"
+	clusterTypes "github.com/pingcap/tipocket/pkg/cluster/types"
+	"github.com/pingcap/tipocket/pkg/core"
+	"github.com/pingcap/tipocket/pkg/history"
+)
+
+// primary key types
+const (
+	PKTypeInt     = "int"
+	PKTypeDecimal = "decimal"
+	PKTypeString  = "string"
+)
+
+// Config is the config of the test case.
+type Config struct {
+	// PKType
+	PKType string
+	// When Partition is set to true, we use multiple partition for multiple account instead of multiple table.
+	Partition bool
+	// When Range is set to true, we use range condition instead of equal condition which
+	// leads to coprocessor request instead of point get request.
+	Range bool
+	// When ReadCommitted is set to true, we use read-committed isolation level for pessimistic transaction.
+	ReadCommitted bool
+	// When UpdateInPlace is set to true, we update with `set balance = balance - @amount`, otherwise we select for update first,
+	// Then update by `set balance = @new_balance`.
+	UpdateInPlace bool
+}
+
+const (
+	vbAccountNum           = 10
+	vbInitialBalance       = 20
+	vbCreateInitialBalance = 10
+)
+
+func (c *Client) genCreateTableSQL(i int) string {
+	var pkType string
+	switch c.cfg.PKType {
+	case PKTypeInt:
+		pkType += "INT"
+	case PKTypeDecimal:
+		pkType += "DECIMAL(10, 2)"
+	case PKTypeString:
+		pkType += "VARCHAR(64)"
+	default:
+		panic("unknown PK type")
+	}
+	var partitionStr string
+	if c.cfg.Partition {
+		partitionStr = " PARTITION BY HASH(id) PARTITIONS 8"
+	}
+	createTableTemplate := `CREATE TABLE %s (
+	id %s PRIMARY KEY,
+	balance DECIMAL(10, 2),
+	balance2 DECIMAL(10, 2) GENERATED ALWAYS AS (-balance) VIRTUAL,
+	created_at TIMESTAMP,
+	updated_at TIMESTAMP,
+	INDEX i_balance (balance),
+	INDEX i_balance2 (balance2),
+	INDEX i_updated_at (updated_at)
+)%s`
+	return fmt.Sprintf(createTableTemplate, c.getTableName(i), pkType, partitionStr)
+}
+
+func (c *Client) genInitialInsertSQL(id int) string {
+	rowID := id
+	if !c.cfg.Partition {
+		rowID = 0
+	}
+	return fmt.Sprintf("insert into %s (id, balance, created_at) VALUES (%s, %d, '%.19s')",
+		c.getTableName(id), c.idValue(rowID), vbInitialBalance, time.Now(),
+	)
+}
+
+// Response ...
+type Response struct {
+	ReqType        VBReqType
+	TS             uint64
+	ReadResult     *BankState
+	CreateResult   *CreateResult
+	DeleteResult   *DeleteResult
+	TransferResult *TransferResult
+	OK             bool
+	Err            string
+}
+
+// IsUnknown implements UnknownResponse interface
+func (r *Response) IsUnknown() bool {
+	return false
```

Comment:
If always return false, how to resolve the situation when client doesn't known whether the request is successful?

Reply:
By checking the transaction status in `v_bank_txn_status` table.